crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -256,19 +267,57 @@ impl<'a> ReflectStruct<'a> {
         &self.meta
     }
 
-    /// Get an iterator over the active fields.
-    pub fn active_fields(&self) -> impl Iterator<Item = &StructField<'a>> {
-        self.fields.iter().filter(|field| !field.attrs.ignore)
+    /// Access the data about which fields should be ignored during serialization.
+    ///
+    /// The returned bitset is a collection of indices obtained from the [`members_to_serialization_blacklist`](crate::utility::members_to_serialization_blacklist) function.
+    #[allow(dead_code)]
+    pub fn serialization_blacklist(&self) -> &BitSet<u32> {
+        &self.serialization_blacklist
     }
 
-    /// Get an iterator over the ignored fields.
-    pub fn ignored_fields(&self) -> impl Iterator<Item = &StructField<'a>> {
-        self.fields.iter().filter(|field| field.attrs.ignore)
+    /// Returns the `GetTypeRegistration` impl as a `TokenStream`.
+    ///
+    /// Returns a specific implementation for structs and this method should be preffered over the generic [`get_type_registration`](crate::ReflectMeta) method
+    pub fn get_type_registration(&self) -> proc_macro2::TokenStream {
+        let reflect_path = self.meta.bevy_reflect_path();
+
+        crate::registration::impl_get_type_registration(
+            self.meta.type_name(),
+            reflect_path,
+            self.meta.traits().idents(),
+            self.meta.generics(),
+            Some(&self.serialization_blacklist),
+        )
     }
 
-    /// Get a collection of all active types.
-    pub fn active_types(&self) -> impl Iterator<Item = &Type> {
-        self.active_fields().map(|field| &field.data.ty)
+    /// Get an iterator over the fields satisfying the given predicate
+    fn fields_by<F: FnMut(&StructField) -> bool>(
```

Comment:
These methods are pretty abstract and not used in many places. I would just inline them and repeat myself.